internal/provisioner/kops_utils.go

```diff
@@ -110,3 +111,49 @@ func getLoadBalancerEndpoint(ctx context.Context, namespace string, logger log.F
 		}
 	}
 }
+
+// GetNGINXLoadBalancerEndpoint returns the load balancer endpoint of the NGINX service.
+func (provisioner *KopsProvisioner) GetNGINXLoadBalancerEndpoint(cluster *model.Cluster, namespace string, logger log.FieldLogger) (string, error) {
```

Comment:
The provisioner is one of the few things that actually has its own logger. I would use that instead of passing one in. This is an example of how you can set up the logger right in the beginning of this method:

```
	logger := provisioner.logger.WithFields(log.Fields{
		"cluster":      cluster.ID,
		"nginx-namespace": namespace,
	})
```

Reply:
Changed to use provisioner logger. Thanks :)